bootstrap/config/config.go

```diff
@@ -283,9 +303,51 @@ func (cp *Processor) Process(
 		}
 	}
 
+	if remoteHosts != nil {
```

Comment:
i am seeing an opportunity that you can absorb this `remoteHosts !=nil` check into your function `applyRemoteHosts`: so that you don't need to add those duplicate != nil check above and here.
The mental model is the same: if remoteHosts != nil means nothing to apply when that applyRemoteHosts() func is called.

Reply:
sorry i meant if remoteHosts == nil

Reply:
Moving this check of `remoteHosts` into `applyRemoteHosts()` func doesn't change the need to check `remoteHosts` in the code above. The check above under `useProvider` is specific to the host for the ConfigProvider (not in configuration) that has to be done before this call so the config is loaded for `applyRemoteHosts`

Reply:
ok, gotcha